python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -196,6 +231,28 @@ def from_assets(
                 auto_materialize_policies_by_key.update(asset.auto_materialize_policies_by_key)
                 backfill_policies_by_key.update({key: asset.backfill_policy for key in asset.keys})
                 code_versions_by_key.update(asset.code_versions_by_key)
+                for key in asset.keys:
+                    execution_types_by_key[key] = (
+                        []
+                        if asset.execution_type == AssetExecutionType.UNEXECUTABLE
+                        else [asset.execution_type]
+                    )
+
+                # Set auto_observe_interval_minutes for external observable assets
+                # This can be removed when/if we have a a solution for mapping
+                # `auto_observe_interval_minutes` to an AutoMaterialzePolicy
+                first_key = next(iter(asset.keys), None)
+                if (
+                    first_key
+                    and SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES
+                    in asset.metadata_by_key[first_key]
+                ):
+                    interval = asset.metadata_by_key[first_key][
+                        SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES
+                    ]
+                    auto_observe_interval_minutes_by_key.update(
+                        {key: interval for key in asset.keys}
+                    )
```

Comment:
this should definitely be a helper function since this is duplicated code from the previous PR AFAICT